src/subdomains/supporting/payment/dto/transaction.dto.ts

```diff
@@ -19,6 +20,45 @@ export enum TransactionState {
   RETURNED = 'Returned',
 }
 
+export enum TransactionError {
+  NA = 'NA',
+  DAILY_LIMIT_EXCEEDED = 'DailyLimitExceeded',
+  ANNUAL_LIMIT_EXCEEDED = 'AnnualLimitExceeded',
+  BANK_ACCOUNT_NOT_MATCHING = 'BankAccountNotMatching',
+  IBAN_CHECK = 'IbanCheck',
+  KYC_REJECTED = 'KycRejected',
+  KYC_MISSING = 'KycMissing',
+  MIN_DEPOSIT_NOT_REACHED = 'MinDepositNotReached',
+  ASSET_NOT_AVAILABLE = 'AssetNotAvailable',
+  STAKING_DISCONTINUED = 'StakingDiscontinued',
+  BANK_NOT_ALLOWED = 'BankNotAllowed',
+  COUNTRY_NOT_ALLOWED = 'CountryNotAllowed',
+  MANUAL_CHECK = 'ManualCheck',
+}
+
+export const TransactionErrorMapper: {
+  [key in AmlReason]: TransactionError;
+} = {
+  [AmlReason.NA]: TransactionError.NA,
+  [AmlReason.NO_COMMUNICATION]: TransactionError.NA,
+  [AmlReason.DAILY_LIMIT]: TransactionError.DAILY_LIMIT_EXCEEDED,
+  [AmlReason.ANNUAL_LIMIT]: TransactionError.ANNUAL_LIMIT_EXCEEDED,
+  [AmlReason.ANNUAL_LIMIT_WITHOUT_KYC]: TransactionError.ANNUAL_LIMIT_EXCEEDED,
+  [AmlReason.USER_DATA_MISMATCH]: TransactionError.BANK_ACCOUNT_NOT_MATCHING,
+  [AmlReason.IBAN_CHECK]: TransactionError.IBAN_CHECK,
+  [AmlReason.KYC_REJECTED]: TransactionError.KYC_REJECTED,
+  [AmlReason.OLKY_NO_KYC]: TransactionError.KYC_MISSING,
+  [AmlReason.NAME_CHECK_WITHOUT_KYC]: TransactionError.KYC_MISSING,
+  [AmlReason.HIGH_RISK_KYC_NEEDED]: TransactionError.KYC_MISSING,
+  [AmlReason.MIN_DEPOSIT_NOT_REACHED]: TransactionError.MIN_DEPOSIT_NOT_REACHED,
+  [AmlReason.ASSET_CURRENTLY_NOT_AVAILABLE]: TransactionError.ASSET_NOT_AVAILABLE,
+  [AmlReason.STAKING_DISCONTINUED]: TransactionError.STAKING_DISCONTINUED,
+  [AmlReason.BANK_NOT_ALLOWED]: TransactionError.BANK_NOT_ALLOWED,
+  [AmlReason.HIGH_RISK_BLOCKED]: TransactionError.BANK_NOT_ALLOWED,
+  [AmlReason.COUNTRY_NOT_ALLOWED]: TransactionError.COUNTRY_NOT_ALLOWED,
+  [AmlReason.MANUAL_CHECK]: TransactionError.MANUAL_CHECK,
```

Comment:
We probably should map AmlReason.NA and AmlReason.MANUAL_CHECK to null? These are no errors.